src/main/java/step2/Calc.java

```diff
@@ -0,0 +1,64 @@
+package step2;
+
+import java.util.Scanner;
+import java.util.Stack;
+import java.util.regex.Pattern;
+
+public class Calc {
+	private final String ONLY_NUMBER = "^[0-9]+$";
+	private final String ONLY_OPERATOR = "^|\\*|\\/|\\+|-|$";
+
+	public static void main(String[] args) {
+		Scanner scanner = new Scanner(System.in);
+		String str = scanner.nextLine();
+		scanner.close();
+
+		System.out.println("result : " + new Calc().getResult(str));
+	}
+
+	public String getResult(String str) {
+		if(this.isEmpty(str)) throw new IllegalArgumentException("입력 값이 null이거나 빈 공백 문자");
+
+		String[] values = str.split(" ");
+		Stack<String> expression = new Stack<>();
+		for(String value : values) {
+			expression.push(value);
+			if(value.matches(ONLY_NUMBER) && expression.size() == 3) {
```

Comment:
현재 String.matches를 통해 연산자 구분을 위한 정규표현식을 사용하고 계시는군요! 
> value.matches(ONLY_NUMBER))

String.matches 내부 구현을 보면 
```
// String.matches
public boolean matches(String regex) {
        return Pattern.matches(regex, this);
}

// Pattern
public static boolean matches(String regex, CharSequence input) {
        Pattern p = Pattern.compile(regex);
        Matcher m = p.matcher(input);
        return m.matches();
}
```
매번 계산시 성능이 비싼 Pattern 인스턴스를 생성하고 Pattern 인스턴스는 한번 쓰고 버려지기 때문에 
성능적으로 좋지 않습니다. 뿐더러 자바 doc에서도 한번 컴파일 후 재사용 할것을 권장하고 있습니다. 

If a pattern is to be used multiple times, compiling it once and reusing
it will be more efficient than invoking this method each time

Pattern 인스턴스를 재사용하도록 캐싱하여 불필요한 객체 생성을 피하도록 개선하면 좋을 것 같아요 :)